worker/upgradedatabase/worker.go

```diff
@@ -100,9 +100,8 @@ func (cfg Config) Validate() error {
 	if cfg.PerformUpgrade == nil {
 		return errors.NotValidf("nil PerformUpgrade function")
 	}
-	a := utils.AttemptStrategy{}
-	if cfg.RetryStrategy == a {
-		return errors.NotValidf("empty RetryStrategy")
+	if cfg.RetryStrategy.Clock == nil || cfg.RetryStrategy.Delay == 0 || (cfg.RetryStrategy.Attempts == 0 && cfg.RetryStrategy.MaxDuration == 0) {
```

Comment:
Can we break this into a validate retry strategy?  As is, it's not apparent which part of the strategy is not valid.